templates/typescript/samples/sample-assistant/src/dialogs/mainDialog.ts

```diff
@@ -2,89 +2,154 @@
+    private onBoardingDialog: OnboardingDialog;
     private switchSkillDialog: SwitchSkillDialog;
+    private skillsConfig: SkillsConfiguration;
     private templateEngine: LocaleTemplateEngineManager;
-    private readonly skillContext: StatePropertyAccessor<SkillContext>;
     private userProfileState: StatePropertyAccessor<IUserProfileState>;
-    private previousResponseAccesor: StatePropertyAccessor<Partial<Activity>[]>
+    private previousResponseAccesor: StatePropertyAccessor<Partial<Activity>[]>;
     
     public constructor(
         settings: IBotSettings,
         services: BotServices,
         templateEngine: LocaleTemplateEngineManager,
         userProfileState: StatePropertyAccessor<IUserProfileState>,
-        skillContext: StatePropertyAccessor<SkillContext>,
         previousResponseAccessor: StatePropertyAccessor<Partial<Activity>[]>,
-        onboardingDialog: OnboardingDialog,
+        onBoardingDialog: OnboardingDialog,
         switchSkillDialog: SwitchSkillDialog,
         skillDialogs: SkillDialog[],
+        skillsConfig: SkillsConfiguration,
         telemetryClient: BotTelemetryClient
     ) {
-        super(MainDialog.name, telemetryClient);
+        super(MainDialog.name);
 
-        this.services = services
-        this.settings = settings
-        this.templateEngine = templateEngine
-        this.telemetryClient = telemetryClient;
+        this.services = services,
+        this.settings = settings,
+        this.templateEngine = templateEngine,
+        this.skillsConfig = skillsConfig,
+        this.telemetryClient = telemetryClient
         
-        // Create user state properties
         this.userProfileState = userProfileState;
-        this.skillContext = skillContext;
-
-        // Create conversation state properties
         this.previousResponseAccesor = previousResponseAccessor;
 
+        const steps: ((sc: WaterfallStepContext) => Promise<DialogTurnResult>)[] = [
+            this.onBoardingStep.bind(this),
+            this.introStep.bind(this),
+            this.routeStep.bind(this),
+            this.finalStep.bind(this)
+        ];
+
+        this.addDialog(new WaterfallDialog (MainDialog.name, steps));
+        this.addDialog(new TextPrompt(TextPrompt.name));
+        this.initialDialogId = MainDialog.name;
+
         // Register dialogs
-        this.onboardingDialog = onboardingDialog
-        this.switchSkillDialog = switchSkillDialog
-        this.addDialog(this.onboardingDialog);
+        this.onBoardingDialog = onBoardingDialog;
+        this.switchSkillDialog = switchSkillDialog;
+        this.addDialog(this.onBoardingDialog);
         this.addDialog(this.switchSkillDialog);
 
+        // Register a QnAMakerDialog for each registered knowledgebase and ensure localised responses are provided.
+        const localizedServices: ICognitiveModelSet  = this.services.getCognitiveModels();
+        localizedServices.qnaConfiguration.forEach( (knowledgebase: QnAMakerDialog): void => {
```

Comment:
```suggestion
        localizedServices.qnaConfiguration.forEach((knowledgebase: QnAMakerDialog): void => {
```

This was marked by ESLint?

Reply:
Yes, qnaConfiguration is not present in ICognitiveModelSet interface

Reply:
I was asking about the space that I removed 😂